### PR TITLE
Route Group Fundamental Sample

### PR DESCRIPTION
**Pull request by fiyazbinhasan**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/26819

@tdykstra I accidentally removed my fork, so this PR https://github.com/dotnet/AspNetCore.Docs/pull/27177 has closed automatically. Creating a new PR. 

However, it would be best if someone from the product team could intervene here. Maybe I'm missing some crucial configuration in order to make the `linkGenerator` work

> These are very different configurations,
> 
> ```
> app.MapGroup("/{accessibility}/todos")
>     .MapTodosApi()
>     .WithTags("Todo Endpoints");
> 
> group.MapGet("/{id}", GetTodo).WithName("GetTodo");
> ```
> 
> With the following approach, you **can't** assign any name to your group. It will throw you an exception saying the name should be globally unique, hence `linkGenerator` won't be able to generate any links.
> 
> ```
> app.MapGroup("/public/todos")
>     .MapTodosApi()
>     .WithTags("Public Todo Endpoints");
> 
> app.MapGroup("/private/todos")
>     .MapTodosApi()
>     .RequireAuthorization();
> ```

In the sample, we should also consider showing a nested route group as well. What do you think?

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@captainsafia Please review, having problem with `linkGenerator`

**Comment by captainsafia** [comment]
@halter73 Do we happen to have a workaround for the link generation issue mentioned in https://github.com/dotnet/AspNetCore.Docs/pull/27177 or should we document it as a known issue?

File: aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group2/Program.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+app.MapGroup("/public/todos")
+    .MapTodosApi()
+    .WithTags("Public Todo Endpoints");
+
+app.MapGroup("/private/todos")
+    .MapTodosApi()
+    .RequireAuthorization();
+
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
```
Comment (captainsafia) [inline aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group2/Program.cs:32]: Can you move this to L18?

Comment (Rick-Anderson) [reply aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group2/Program.cs:32]: @captainsafia 
> Can you move this to L18?
Is this what you want?

```suggestion
var app = builder.Build();

app.MapGroup("/private/todos")
    .MapTodosApi()
    .RequireAuthorization();

using var scope = app.Services.CreateScope();
var db = scope.ServiceProvider.GetService<TodoGroupDbContext>();
db?.Database.MigrateAsync();

app.MapGroup("/public/todos")
    .MapTodosApi()
    .WithTags("Public Todo Endpoints");

// Configure the HTTP request pipeline.
if (app.Environment.IsDevelopment())
```

**Comment by Rick-Anderson** [comment]
@fiyazbinhasan is link generation working now?

**Comment by halter73** [comment]
> With the following approach, you can't assign any name to your group. It will throw you an exception saying the name should be globally unique, hence linkGenerator won't be able to generate any links.

Groups doesn't give you a free lunch when it comes to naming endpoints. Endpoints defined by `MapGet`, `MapPost`, etc... don't work with special `RouteValuesAddress`-based extension methods like `GetPathByPage` and `GetPathByAction`. It might be interesting to make these extension methods and make the key the combined `RoutePattern.RawText` (`GetPathByRoutePattern`?), but that's not something we've implemented yet.

As it stands, `GetPathByName` does work with `MapGet`, `MapPost`, etc... but that requires uniquely naming each endpoint using `WithName` or the `EndpointNameAttribute` which doesn't make sense on a group. This isn't a group limitation. This is just a current limitation of minimal route handlers. For now, I don't think we should demonstrate link generation in a group-focused sample.

> Do we happen to have a workaround for the link generation issue mentioned in #27177 or should we document it as a known issue?

It turns out that the problem with that test was not related to groups after all. Even without the MapGroup calls, the page failed to generate a contact link. I fixed the test. https://github.com/dotnet/aspnetcore/pull/44562

**Review by tdykstra — APPROVED** [review]
Thanks, @fiyazbinhasan

**Pull request merged**